src/home_robot/home_robot/navigation_policy/language_navigation/languagenav_frontier_exploration_policy.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+import numpy as np
+import scipy
+import skimage.morphology
+import torch
+import torch.nn as nn
+from sklearn.cluster import DBSCAN
+
+from home_robot.mapping.semantic.constants import MapConstants as MC
+from home_robot.utils.morphology import binary_dilation
+
+
+class LanguageNavFrontierExplorationPolicy(nn.Module):
+    """
+    Policy to select high-level goals for Object Goal Navigation:
+    go to object goal if it is mapped and explore frontier (closest
+    unexplored region) otherwise.
+    """
+
+    def __init__(self, exploration_strategy: str):
+        super().__init__()
+        assert exploration_strategy in ["seen_frontier", "been_close_to_frontier"]
+        self.exploration_strategy = exploration_strategy
+
+        self.dilate_explored_kernel = nn.Parameter(
+            torch.from_numpy(skimage.morphology.disk(10))
+            .unsqueeze(0)
+            .unsqueeze(0)
+            .float(),
+            requires_grad=False,
+        )
+        self.select_border_kernel = nn.Parameter(
+            torch.from_numpy(skimage.morphology.disk(1))
+            .unsqueeze(0)
+            .unsqueeze(0)
+            .float(),
+            requires_grad=False,
+        )
+
+    @property
+    def goal_update_steps(self):
+        return 1
+
+    def reach_single_category(self, map_features, category, reject_visited_targets):
+        # if the goal is found, reach it
+        goal_map, found_goal = self.reach_goal_if_in_map(
+            map_features, category, reject_visited_targets=reject_visited_targets
+        )
+        # otherwise, do frontier exploration
+        goal_map = self.explore_otherwise(map_features, goal_map, found_goal)
+        return goal_map, found_goal
+
+    def forward(
+        self,
+        map_features,
+        object_category=None,
+        reject_visited_targets=False,
+    ):
+        """
+        Arguments:
+            map_features: semantic map features of shape
+             (batch_size, 9 + num_sem_categories, M, M)
+            object_category: object goal category
+        Returns:
+            goal_map: binary map encoding goal(s) of shape (batch_size, M, M)
+            found_goal: binary variables to denote whether we found the object
+            goal category of shape (batch_size,)
+        """
+        assert object_category is not None
+
+        # Here, the goal is specified by a single object
+        return self.reach_single_category(
+            map_features, object_category, reject_visited_targets
+        )
+
+    def cluster_filtering(self, m):
```

Comment:
Is this going to be shared across different files? @yvsriram  added cluster filtering for the world state right?